lisa/sut_orchestrator/libvirt/ch_platform.py

```diff
@@ -176,6 +176,22 @@ def _create_node_os_disk(
         self, environment: Environment, log: Logger, node: Node
     ) -> None:
         node_context = get_node_context(node)
+        if environment.runbook.nodes_requirement:
+            node_req: List[schema.NodeSpace] = environment.runbook.nodes_requirement
+            for _, node_space in enumerate(node_req):
```

Comment:
I don't think you should be running the resize operation multiple times.

Reply:
Here we will run only for given requirements. It will only run for once if it is the case in runbook.

Reply:
I don't understand what you are saying. `NodeSpace` is only really used for provisioning the node. It shouldn't be used here. And it most definitely should not be used in a loop per node.

Reply:
i used node_context  now as you suggested.